### PR TITLE
refactor(BA-3485): Integrate `CreateScalingGroup` action with existing GQL resolver

### LINKED ISSUE #7485: Integrate CreateScalingGroup Action with existing GraphQL implementation
## Objective

Complete Create action with GraphQL integration and comprehensive tests.

## Implementation Details

- GraphQL mutation modification (~30 lines)
  - `CreateScalingGroupInput` → `Creator` conversion logic
  - `simple_db_mutate_returning_item` → `processors.scaling_group.create.wait_for_complete()`
- ScalingGroupOpts conversion logic cleanup (~20 lines)
- Additional Unit tests:
  - `test_create_scaling_group_service.py` (~80 lines)
  - `test_create_scaling_group_repository.py` (~100 lines)
- Edge case tests (duplicate name, validation, etc.) (~50 lines)

## Estimated Lines

~280 lines

## Dependencies

- Story 3: CreateScalingGroup Action - Creator Spec

## Acceptance Criteria

- [ ] GraphQL mutation uses new processor pattern
- [ ] Input conversion properly handles all fields including ScalingGroupOpts
- [ ] Edge cases tested (duplicate names, invalid inputs)
- [ ] All unit tests pass
- [ ] Legacy `simple_db_mutate_returning_item` removed from CreateScalingGroup

JIRA Issue: BA-3485

### PR DESCRIPTION
**Pull request by jopemachine**
resolves #7485 (BA-3485)

**Checklist:** (if applicable)

- [ ] Milestone metadata specifying the target backport version
- [x] Mention to the original issue
- [ ] Installer updates including:
  - Fixtures for db schema changes
  - New mandatory config options
- [ ] Update of end-to-end CLI integration tests in `ai.backend.test`
- [ ] API server-client counterparts (e.g., manager API -> client SDK)
- [x] Test case(s) to:
  - Demonstrate the difference of before/after
  - Demonstrate the flow of abstract/conceptual models with a concrete implementation
- [ ] Documentation
  - Contents in the `docs` directory
  - docstrings in public interfaces and type annotations

### CODE REVIEW

File: src/ai/backend/manager/data/scaling_group/types.py [hunk 1]
```diff
@@ -61,6 +61,19 @@ class ScalingGroupSchedulerOptions:
...
+        return {
+            "allowed_session_types": [st.value for st in self.allowed_session_types],
+            "pending_timeout": self.pending_timeout.total_seconds(),
+            "config": dict(self.config),
+            "agent_selection_strategy": self.agent_selection_strategy.value,
+            "agent_selector_config": dict(self.agent_selector_config),
+            "enforce_spreading_endpoint_replica": self.enforce_spreading_endpoint_replica,
+            "allow_fractional_resource_fragmentation": self.allow_fractional_resource_fragmentation,
+            "route_cleanup_target_statuses": self.route_cleanup_target_statuses,
+        }
```
Comment (HyeockJinKim) [inline src/ai/backend/manager/data/scaling_group/types.py:75]: Could you please write test code because there were frequent issues in the transformation logic?
